### PR TITLE
Adds `clearUnapprovedTxs` function to `TransactionController`

### PR DESCRIPTION
**Pull request by OGPoyraz**
## Explanation

This PR adds `clearUnapprovedTxs` function to `TransactionController` which is needed in the extension client.

### `@metamask/transaction-controller`

- **Added**: Add `clearUnapprovedTxs` public function to `TransactionController` 

## Checklist

- [X] I've updated the test suite for new or updated code as appropriate
- [X] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [X] I've highlighted breaking changes using the "BREAKING" category above as appropriate

### CODE REVIEW

File: packages/transaction-controller/src/TransactionController.ts [hunk 1]
```diff
@@ -1235,6 +1235,16 @@ export class TransactionController extends BaseController<
     return this.nonceTracker.getNonceLock(address);
   }
 
+  /**
+   * Filters out the unapproved transactions from state
+   */
+  clearUnapprovedTxs() {
```
Comment (matthewwalsh0) [inline packages/transaction-controller/src/TransactionController.ts:1241]: Shall we use the full form so `clearUnapprovedTransactions` to match all our types which are now using `Transaction`?

File: packages/transaction-controller/src/TransactionController.ts [hunk 2]
```diff
@@ -1235,6 +1235,16 @@ export class TransactionController extends BaseController<
     return this.nonceTracker.getNonceLock(address);
   }
 
+  /**
+   * Filters out the unapproved transactions from state
```
Comment (matthewwalsh0) [inline packages/transaction-controller/src/TransactionController.ts:1239]:
```suggestion
   * Removes unapproved transactions from state.
```

File: packages/transaction-controller/src/TransactionController.test.ts [hunk 3]
```diff
@@ -2449,6 +2449,43 @@ describe('TransactionController', () => {
...
   });
 
+  describe('clearUnapprovedTxs', () => {
+    it('clears unapproved transactions', async () => {
+      const controller = newController();
+
+      const unapprovedTxId = '1';
+      const confirmedTxId = '2';
+
+      controller.state.transactions.push({
```
Comment (matthewwalsh0) [inline packages/transaction-controller/src/TransactionController.test.ts:2459]: Should we add a second entry to bulk methods like this to ensure it's iterating correctly?

Comment (OGPoyraz) [reply packages/transaction-controller/src/TransactionController.test.ts]: Fixed

File: packages/transaction-controller/src/TransactionController.ts (hunk 1 above)
Comment (OGPoyraz) [reply packages/transaction-controller/src/TransactionController.ts]: Sure, I didn't want to change name of it but make sense

**Review by matthewwalsh0 — APPROVED** [review]

**Pull request merged**